File: Day51/ConsumeService/Services/UserService.cs
using ConsumeService.Interfaces;
using ConsumeService.Models;
using System.Net.Http.Headers;


namespace ConsumeService.Services
{
    public class UserService : IUser
    {
       HttpClient _httpClient;

        public UserService()
        {
            _httpClient = new HttpClient();
            _httpClient.DefaultRequestHeaders.Accept.Add(new MediaTypeWithQualityHeaderValue("application/json"));
        }
        public async Task<User> Login(User user)
        {
            var response = await _httpClient.PostAsJsonAsync("http://localhost:5087/api/User/Login", user);
            if (response.IsSuccessStatusCode)
            {
                var data = await response.Content.ReadAsAsync<User>();
                return data;
            }
            return null;
        }
    }
}

File: Day54/pizzaapp/src/app/services/user.service.ts
import { HttpClient } from "@angular/common/http";
import { User } from "../models/user";
import { Injectable } from "@angular/core";
import { Observable } from "rxjs";

@Injectable()
export class UserService{

    constructor(private http:HttpClient){

    }


    login(user:User):Observable<any>{
        console.log(user);
        return this.http.post("http://localhost:5087/api/User/Login",user);
    }
}